post-processing/post_processing.py

```diff
@@ -104,8 +104,35 @@ def run_post_processing(self, log_path, config):
         # apply user-specified types to all relevant columns
         for col in all_columns:
             if config["column_types"].get(col):
-                # FIXME: this can and will raise TypeErrors, some of which should be dealt with internally
-                df[col] = df[col].astype(config["column_types"][col])
+
+                # get user input type
+                conversion_type = config["column_types"][col]
+                # allow user to specify "datetime" as a type (internally convert to "datetime64")
+                conversion_type += "64" if conversion_type == "datetime" else ""
+
+                # internal type conversion
+                if pd.api.types.is_string_dtype(conversion_type):
+                    # all strings treated as object (nullable)
+                    conversion_type = "object"
+                elif pd.api.types.is_float_dtype(conversion_type):
+                    # all floats treated as float64 (nullable)
+                    conversion_type = "float64"
+                elif pd.api.types.is_integer_dtype(conversion_type):
+                    # all integers treated as Int64 (nullable)
+                    # note: default pandas integer type is int64 (not nullable)
+                    conversion_type = "Int64"
+                elif pd.api.types.is_datetime64_any_dtype(conversion_type):
+                    # all datetimes treated as datetime64[ns] (nullable)
+                    conversion_type = "datetime64[ns]"
+                else:
+                    raise RuntimeError("Unsupported user-specified type '{0}' for column '{1}'.".format(conversion_type, col))

```

Comment:
Could also add a boolean type here. Don't think I've seen any perflogs that actually use bools, but pandas does have a nullable boolean type (missing values are treated as False).

Reply:
Don't worry about it for now, we can add `bool`s later if needed.